mmdet3d/datasets/nuscenes_mono_dataset.py

```diff
@@ -519,18 +526,129 @@ def evaluate(self,
             tmp_dir.cleanup()
 
         if show:
-            self.show(results, out_dir)
+            self.show(results, out_dir, pipeline=pipeline)
         return results_dict
 
-    def show(self, results, out_dir):
+    @staticmethod
+    def _get_data(results, key):
```

Comment:
In other 3D datasets, we have these functions to enable data loading from `eval_pipeline`. However, as `NusMono` inherited from 2D datasets, I have to duplicate these functions here. Kind of ugly I think.

Reply:
Can we move it out of a specific class as a single utilized function if most of them in different classes are very similar?

Reply:
Yes. Will do this in a new PR.